modules/dnn/include/opencv2/dnn/dnn.hpp

```diff
@@ -1281,6 +1281,96 @@ CV__DNN_INLINE_NS_BEGIN
+          * @param[out] results: A set of text recognition results.
+          * @param[in] roiPolygons: (Optional) A set of text detection results. If it is not empty, all the ROIs will be cropped as the network inputs.
+          */
+         CV_WRAP void recognize(InputArray frame, String decodeType, CV_OUT std::vector<String> & results,
+                                const std::vector<std::vector<Point>> & roiPolygons = std::vector<std::vector<Point>>());
+     protected:
+         std::vector<String> vocabulary;
+
+         /**
+          * @brief decode the output of the network.
+          * @param[in] prediction: The output of the network.
+          * @param[in] decodeType: The decoding method of translating the network output into string.
+          * @returns a string holding the decoded recognition result
+          */
+         CV_WRAP String decode(Mat prediction, String decodeType);
+     };
+
+     /** @brief This class represents high-level API for text detection networks.
+      *
+      * TextDetectionModel allows to set params for preprocessing input image.
+      * TextDetectionModel creates net from file with trained weights and config,
+      * sets preprocessing input, runs forward pass and return detection results.
+      * For TextDetectionModel, "DB" is supported.
+      */
+     class CV_EXPORTS_W TextDetectionModel : public Model
+     {
+     public:
+         /**
+          * @brief Create text detection model from network represented in one of the supported formats.
+          * An order of @p model and @p config arguments does not matter.
+          * @param[in] model Binary file contains trained weights.
+          * @param[in] config Text file contains network configuration.
+          */
+         CV_WRAP TextDetectionModel(const String& model, const String& config = "");
+
+         /**
+          * @brief Create model from deep learning network.
+          * @param[in] network Net object.
+          */
+         CV_WRAP TextDetectionModel(const Net& network);
+
+         /**
+          * @brief Given the @p input frame, create input blob, run net and return detection results. For more information about the hyper-parameters, please refer to https://github.com/MhLiao/DB.
+          * @param[in] frame: The input image.
+          * @param[out] results: A set of text detection results.
+          * @param[in] outputType: The output shape type: {0: multi-oriented quadrilateral; 1: polygon}
+          * @param[in] binThresh: The threshold of the binary map. It can be set in [0, 1.0], and is usually set to 0.3 in the official code.
+          * @param[in] polyThresh: The threshold of text polygons. It can be set in [0, 1.0], and is usually set to 0.5, 0.6, and 0.7 in the official code.
+          * @param[in] unclipRatio: The unclip ratio of the detected text region, which determines the output size. It is set to 2.0 in the official code.
```

Comment:
what does it mean "official code"?

Reply:
I have put the official code's URL of this algorithm at the top of these comments.